src/pages/index.js

```diff
@@ -46,41 +72,154 @@ const popupAddForm = new PopupWithForm(
 popupAddForm.setEventListeners();
 
 const popupImage = new PopupWithImage("#modal__image");
-popupImage.setEventListiners();
+popupImage.setEventListeners();
 
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
 // FUNCTIONS/////
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getCardElement();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleLikeClicks,
+    handleDeleteSubmit
+  );
+  return card.getView();
 }
 
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+function handleImageClick(cardData) {
+  popupImage.open(cardData);
+}
+
+const formValidators = {};
+const forms = document.querySelectorAll(".modal__form");
+forms.forEach((form) => {
+  const formValidator = new FormValidator(config, form);
+  formValidators[form.id] = formValidator;
+  formValidator.enableValidation();
+});
+
+const handleAvatarSubmit = ({ avatar }) => {
+  editAvatar.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```